src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,289 @@
+                    BlockPos row4 = new BlockPos(x3, newY, i);
+                    world.setBlockState(row4, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.WEST));
+                }
+            }
+            newY = checkIfPlaceable(x1,y,z1,world);
+            BlockPos corner1 = new BlockPos(x1, newY, z1);
+            newY = checkIfPlaceable(x1,y,z3,world);
+            BlockPos corner2 = new BlockPos(x1, newY, z3);
+            newY = checkIfPlaceable(x3,y,z1,world);
+            BlockPos corner3 = new BlockPos(x3, newY, z1);
+            newY = checkIfPlaceable(x3,y,z3,world);
+            BlockPos corner4 = new BlockPos(x3, newY, z3);
+            world.setBlockState(corner1, ModBlocks.blockConstructionTapeC.getDefaultState().withProperty(FACING, EnumFacing.SOUTH));
+            world.setBlockState(corner2, ModBlocks.blockConstructionTapeC.getDefaultState().withProperty(FACING, EnumFacing.EAST));
+            world.setBlockState(corner3, ModBlocks.blockConstructionTapeC.getDefaultState().withProperty(FACING, EnumFacing.WEST));
+            world.setBlockState(corner4, ModBlocks.blockConstructionTapeC.getDefaultState().withProperty(FACING, EnumFacing.NORTH));
+        }
+
+
+
+    public static void removeConstructionTape(@NotNull AbstractBuilding building, @NotNull World world)
+    {
+        removeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    public static void removeConstructionTape(@NotNull WorkOrderBuild workOrder,@NotNull World world)
+    {
+        final StructureWrapper wrapper = new StructureWrapper(world, (workOrder.getStructureName()));
+        final BlockPos pos = workOrder.getBuildingLocation();
+        int tempRotation = 0;
+        if (workOrder.getRotation() == 0 && !(workOrder instanceof WorkOrderBuildDecoration))
+        {
+            final IBlockState blockState = world.getBlockState(pos);
+            if (blockState.getBlock() instanceof AbstractBlockHut)
+            {
+                tempRotation = BlockUtils.getRotationFromFacing(blockState.getValue(AbstractBlockHut.FACING));
+            }
+        }
+        else
+        {
+            tempRotation = workOrder.getRotation();
+        }
+        wrapper.rotate(tempRotation, world, workOrder.getBuildingLocation());
+        wrapper.setPosition(pos);
+        int x1 = wrapper.getPosition().getX() - wrapper.getOffset().getX() - 1;
+        int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - 1;
+        int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX());
+        int z3 = wrapper.getPosition().getZ() + (wrapper.getLength() - wrapper.getOffset().getZ());
+        if (x1 < x3)
+        {
+            for (int i = x1; i <= x3; i++)
+            {
+                for (int y = 1; y <= 255; y++)
+                {
+                    BlockPos row1 = new BlockPos(i, y, z1);
+                    BlockPos row2 = new BlockPos(i, y, z3);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'row2' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)